pkg/monitor/cluster/debugpods.go

```diff
@@ -0,0 +1,72 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	oc "github.com/openshift/oc/pkg/helpers/newapp/app"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+// Each cluster is being checked every 1 minute. emitDebugPodsCount
+// lists all events (age < 70 seconds) in the default namespace
+// in order to detect recent events of debug-pod creation.
+func (mon *Monitor) emitDebugPodsCount(ctx context.Context) error {
+	m := map[string]string{
+		"reason":              "Started",
+		"involvedObject.kind": "Pod",
+		// oc debug node creates a pod with a "container-00"
+		// container name.
+		"involvedObject.fieldPath": "spec.containers{container-00}",
+	}
+	lo := metav1.ListOptions{
+		FieldSelector: fields.SelectorFromSet(m).String(),
+	}
+	events, err := mon.cli.CoreV1().Events("default").List(ctx, lo)
+	if err != nil {
+		return err
+	}
+
+	nodes, err := mon.cli.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	podNames := getDebugPodNames(nodes.Items)
+
+	count := countDebugPods(podNames, events.Items)
+	mon.emitGauge("debugpods.count", int64(count), nil)
+
+	return nil
+}
+
+func countDebugPods(debugPods []string, events []corev1.Event) int {
+	count := 0
+	for _, e := range events {
+		if eventIsNew(e) && stringutils.Contains(debugPods, e.InvolvedObject.Name) {
+			count++
+		}
+	}
+	return count
+}
+
+func eventIsNew(event corev1.Event) bool {
+	// corev1.Event.LastTimeStamp is deprecated in newer k8s releases.
+	// Update this check when ARO's k8s version is upgraded
+	return time.Since(event.LastTimestamp.Time) < time.Second*120
+}
+
+func getDebugPodNames(nodes []corev1.Node) []string {
+	names := make([]string, len(nodes))
+	for i, n := range nodes {
+		names[i] = fmt.Sprintf("%s-debug", oc.MakeSimpleName(n.Name))
```

Comment:
Do I understand correctly, we use `oc` only for SimpleName?
If yes, how would you solve the problem without `oc`?
I would rather not vendor in so many dependencies just for this tiny function. Thank you.

Reply:
Yeah, we need it only for this function.
This problem can be solved by looping through all pod names and checking the last 6 symbols of their names (`-debug`). It is not a super reliable solution, but it will work unless users try to confuse us consciously.
I agree that importing this library is overkill. Are you ok with the idea I proposed?